src/Donations/RecordGoogleEventWithGA4OnFrontend.php

```diff
@@ -250,6 +250,82 @@ public function handleGiveEmbedFooter()
+     */
+    function handleFrontendEventsOnV3Forms() {
+        if (!$this->canPrintScript()) {
+            return;
+        }
+
+        $tracking_id   = $this->settingRepository->getGoogleAnalytics4WebStreamMeasurementId();
+        $affiliation   = $this->settingRepository->getTrackAffiliation();
+        $trackCategory = $this->settingRepository->getTrackCategory();
+        $trackListName = $this->settingRepository->getTrackListName();
+        ?>
+
+        <script async src="https://www.googletagmanager.com/gtag/js?id=<?esc_js($tracking_id)?>"></script>
+        <script>
+            window.dataLayer = window.dataLayer || [];
+            function gtag(){dataLayer.push(arguments)}
+            gtag('js', new Date());
+            gtag('config', '<?php echo esc_js($tracking_id)?>',{
+                'page_title': window.parent.document.title
+            });
+
+
+            window.addEventListener('DOMContentLoaded', function() {
+                var form = document.querySelector('#give-next-gen');
+                var defaultDonationAmount = form.querySelector('input[name="amount"]').value;
+                var formId = form.querySelector('[name="formId"]').value;
+
+                var {currency, name} =  window.givewpDonationFormExports.form;
+
+                gtag('event', 'view_item', {
+                    currency: currency,
+                    value: defaultDonationAmount,
+                    items: [
+                        {
+                            item_id: formId,
+                            item_name: name,
+                            item_brand: 'Fundraising',
+                            affiliation: '<? echo esc_js($affiliation)?>',
+                            item_category: `<? echo esc_js($trackCategory)?>`,
+                            item_list_name: `<? echo esc_js($trackListName)?>`,
+                        }
+                    ]
+                })
+
+                form.addEventListener('submit', function(){
+                    var donationAmount = form.querySelector('input[name="amount"]').value;
+                    var gateway = form.querySelector('input[name="gatewayId"]:checked').value;
+                    var donationType = form.querySelector('input[name="donationType"]').value;
+                    var selectedCurrency = form.querySelector('input[name="currency"]').value;
+
+                    gtag('event', 'begin_checkout', {
+                        currency: selectedCurrency,
+                        value: donationAmount,
+                        items: [
+                            {
+                                item_id: formId,
+                                item_name: name,
+                                item_brand: 'Fundraising',
+                                affiliation: '<? echo esc_js($this->settingRepository->getTrackAffiliation())?>',
+                                item_category: '<? echo esc_js($this->settingRepository->getTrackCategory())?>',
```

Comment:
Is this a duplication of `$trackCategory` from above? Could we use the `$trackCategory` variable here instead of calling from the Settings Repository again?

Reply:
02879a8577e72b1bdf42e663ae9fb6aeb96514f3